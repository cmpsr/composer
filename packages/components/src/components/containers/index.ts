export * from './Box';
export * from './Grid';
export * from './Section';
export * from './SectionChild';
