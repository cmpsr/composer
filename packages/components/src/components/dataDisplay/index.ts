export { ListItem } from '@chakra-ui/react';
export * from './Badge';
export * from './Divider';
export * from './NumberedList';
export * from './Symbol';
export * from './Tag';
export * from './UnorderedList';
