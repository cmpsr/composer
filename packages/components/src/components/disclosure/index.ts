export * from './Accordion';
export * from './Tabs';
