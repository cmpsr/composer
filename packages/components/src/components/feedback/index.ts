export * from './Alert';
export * from './InlineAlert';
export * from './ProgressBar';
export * from './ProgressCircular';
export * from './Skeleton';
export * from './Spinner';
export * from './Toast';
