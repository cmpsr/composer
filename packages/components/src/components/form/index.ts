export * from './Autocomplete';
export * from './AutocompleteMultiSelect';
export * from './Button';
export * from './ButtonIcon';
export * from './Checkbox';
export * from './DateInput';
export * from './IconButton';
export * from './Input';
export * from './MaskInput';
export * from './NumberInput';
export * from './PhoneInput';
export * from './PinInput';
export * from './Radio';
export * from './RangeSlider';
export * from './SegmentedButton';
export * from './Select';
export * from './Slider';
export * from './Switch';
export * from './Textarea';
