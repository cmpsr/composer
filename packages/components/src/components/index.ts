export * from './layouts';
export * from './primitives';
export * from './navigation';
export * from './notifications';
