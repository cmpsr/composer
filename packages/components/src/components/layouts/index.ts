export * from './Box';
export * from './Flex';
export * from './Grid';
export * from './Section';
