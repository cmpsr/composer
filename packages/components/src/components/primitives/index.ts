export * from './Accordion';
export * from './Alert';
export * from './Avatar';
export * from './Badge';
export * from './Breadcrumb';
export * from './Button';
export * from './Card';
export * from './CircularProgress';
export * from './Checkbox';
export * from './Divider';
export * from './Image';
export * from './Icons';
export * from './Input';
export * from './Link';
export * from './Progress';
export * from './Radio';
export * from './Select';
export * from './Slider';
export * from './Spinner';
export * from './Switch';
export * from './Tabs';
export * from './Tag';
export * from './Text';
export * from './TextPairing';
export * from './Textarea'
