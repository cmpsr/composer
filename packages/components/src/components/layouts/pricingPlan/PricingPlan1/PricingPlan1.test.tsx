import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { PricingPlan1, PRICING_PLAN_DEFAULT_TEST_ID } from '.';
import { IMAGE_DEFAULT_TEST_ID } from 'components/primitives';

describe('PricingPlan1', () => {
  const title = 'test';
  const price = '10';
  const items = ['test 1', 'test 2', 'test 3', 'test 4'];
  const button = <a>test</a>;

  it('should match snapshot', () => {
    const pricingPlan = renderer.create(
      <PricingPlan1 title={title} price={price}>
        {{ listItems: items, button: button }}
      </PricingPlan1>,
    );
    const tree = pricingPlan.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render component', () => {
    const { getByTestId } = render(
      <PricingPlan1 title={title} price={price} />,
    );
    const pricingPlan = getByTestId(PRICING_PLAN_DEFAULT_TEST_ID);
    expect(pricingPlan).toBeInTheDocument();
  });
  it('should render logo', () => {
    const { getByTestId } = render(
      <PricingPlan1 title={title} price={price} />,
    );
    const pricingPlan = getByTestId(IMAGE_DEFAULT_TEST_ID);
    expect(pricingPlan).toBeInTheDocument();
  });
  it('should render className', () => {
    const { getByTestId } = render(
      <PricingPlan1 title={title} price={price} className="foo" />,
    );
    const pricingPlan = getByTestId(PRICING_PLAN_DEFAULT_TEST_ID);
    expect(pricingPlan.classList).toContain('foo');
  });
  it('should render title', () => {
    const { getByText } = render(<PricingPlan1 title={title} price={price} />);
    const pricingPlan = getByText('test');
    expect(pricingPlan).toBeInTheDocument();
  });
  it('should render price', () => {
    const { getByText } = render(<PricingPlan1 title={title} price={price} />);
    const pricingPlan = getByText('10');
    expect(pricingPlan).toBeInTheDocument();
  });
  it('should render children list', () => {
    const { getByText } = render(
      <PricingPlan1 title={title} price={price}>
        {{ listItems: items }}
      </PricingPlan1>,
    );
    const pricingPlan = getByText('test 1');
    expect(pricingPlan).toBeInTheDocument();
  });
  it('should render children button', () => {
    const childrenTestId = 'Children-TestId';
    const link = <a data-testid={childrenTestId}>test</a>;
    const { getByTestId } = render(
      <PricingPlan1 title={title} price={price}>
        {{ listItems: items, button: link }}
      </PricingPlan1>,
    );
    const pricingPlan = getByTestId(childrenTestId);
    expect(pricingPlan).toBeInTheDocument();
  });
});
