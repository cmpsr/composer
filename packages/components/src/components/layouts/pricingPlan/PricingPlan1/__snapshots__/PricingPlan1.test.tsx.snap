// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PricingPlan1 should match snapshot 1`] = `
<div
  className="rounded shadow-lg px-6 py-8"
  data-testid="pricing_plan1"
>
  <div
    className="flex justify-between items-center"
  >
    <h5
      className="typo-headline-5 font-bold"
      data-testid="typography"
    >
      test
    </h5>
    <a
      className="flex-grow-0 text-btn"
      data-testid="link"
      href="https://www.google.com"
    >
      <picture
        className="select-none"
      >
        <source
          className="select-none h-10"
          data-testid="image"
          srcSet="https://images.ctfassets.net/o9153kt66j4s/6jk4DaIRMlx1CyOXIiLN5F/acebc0468bf2df33d069774c8e8e5bf6/Composer_-_white.svg?fm=webp"
          type="image/webp"
        />
        <img
          alt=""
          className="select-none h-10"
          src="https://images.ctfassets.net/o9153kt66j4s/6jk4DaIRMlx1CyOXIiLN5F/acebc0468bf2df33d069774c8e8e5bf6/Composer_-_white.svg"
        />
      </picture>
    </a>
  </div>
  <div
    className="flex flex-row items-baseline"
  >
    <span
      className="typo-headline-5 font-bold self-start mt-2"
      data-testid="typography"
    >
      $
    </span>
    <span
      className="typo-headline-1"
      data-testid="typography"
    >
      10
    </span>
    <span
      className="typo-headline-6 ml-2"
      data-testid="typography"
    >
      / Mo
    </span>
  </div>
  <div
    className="mt-6 mb-10"
  >
    <ul>
      <li
        className="mb-3"
      >
        <svg
          className="inline-block mr-5"
          data-testid="icon"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
        >
          <path
            d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
        <span
          className="typo-body text-gray-500"
          data-testid="typography"
        >
          test 1
        </span>
      </li>
      <li
        className="mb-3"
      >
        <svg
          className="inline-block mr-5"
          data-testid="icon"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
        >
          <path
            d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
        <span
          className="typo-body text-gray-500"
          data-testid="typography"
        >
          test 2
        </span>
      </li>
      <li
        className="mb-3"
      >
        <svg
          className="inline-block mr-5"
          data-testid="icon"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
        >
          <path
            d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
        <span
          className="typo-body text-gray-500"
          data-testid="typography"
        >
          test 3
        </span>
      </li>
      <li
        className="mb-3"
      >
        <svg
          className="inline-block mr-5"
          data-testid="icon"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
        >
          <path
            d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
        <span
          className="typo-body text-gray-500"
          data-testid="typography"
        >
          test 4
        </span>
      </li>
    </ul>
  </div>
  <a>
    test
  </a>
</div>
`;
